### PR TITLE
Encoding uint pointers

### PR DESCRIPTION
**Pull request by poszu**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,23 +2,27 @@
...
+| string      | same as []byte                                                     |
+| [...]byte   | appended to the result                                             |
+| bool        | 1 byte, 0 for false, 1 for true                                    |
+| Object{}    | concatenation of fields                                            |
+| *Object{}   | Option. 0 for nil, 1 for Object{}. if 1 - decode Object{}          |
+| uint8       | compact u8 [TODO no need for compact u8]                           |
+| uint16      | compact u16                                                        |
+| uint32      | compact u32                                                        |
+| uint34      | compact u64                                                        |
+| *uint8      | Option (0 for nil, 1 otherwise), followed by fixed-width u8        | `&255 -> 01FF`                                        |
```
Comment (dshulyak) [inline README.md:17]: inconsistent with u8. i know that there is a TODO, but it probably can't be changed now without breaking previous encoding

File: README.md [hunk 2]
```diff
@@ -2,23 +2,27 @@
...
+| *uint16     | Option (0 for nil, 1 otherwise), followed by compact u16           | `&255 -> 01FD03`                                      |
+| *uint32     | Option (0 for nil, 1 otherwise), followed by compact u32           | `&255 -> 01FD03`                                      |
+| *uint34     | Option (0 for nil, 1 otherwise), followed by compact u64           | `&255 -> 01FD03`                                      |
```
Comment (dshulyak) [inline README.md:20]: typo . uint64

**Review by dshulyak — APPROVED** [review]
do you really have a use case for encoding options with integers where `zero` integer can't be used instead?

**Comment by poszu** [comment]
> do you really have a use case for encoding options with integers where `zero` integer can't be used instead?

Yes, for example, a vrf nonce index, for which a zero value is also a valid value.

File: README.md (hunk 1 above)
Comment (poszu) [reply README.md:17]: I know, but encoding a U8 in a compact form doesn't make any sense AFAIR as it's slower and larger for values > 63. I'm not sure what is better here: being consistent or having the better implementation for u8. Wdyt?

Comment (dshulyak) [reply README.md:17]: i would probably keep it consistent, as in golang you also have `byte` that can be used to "encode" in full form. but it feels inconsequential so any option is fine

File: encoder_test.go [hunk 3]
```diff
@@ -129,6 +129,71 @@ func uint64TestCases() []compactTestCase[uint64] {
 	}
 }
 
+func newInteger[T ~uint8 | ~uint16 | ~uint32 | ~uint64](v T) *T {
```
Comment (fasmat) [inline encoder_test.go:132]: NIT: name could be more explanatory:

```suggestion
func intPtr[T ~uint8 | ~uint16 | ~uint32 | ~uint64](v T) *T {
```

**Review by fasmat — APPROVED** [review]

**Review by fasmat — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (poszu) [reply README.md]: I changed to encode `*uint8` as compact for consistency. In the future, let's add a way to change the behavior like a `scale:fixed` tag.

**Pull request merged**